ferry.go

```diff
@@ -324,6 +371,26 @@ func (f *Ferry) Run() {
 		handleError("throttler", f.Throttler.Run(ctx))
 	}()
 
+	if f.DumpStateToStdoutOnSignal {
+		supportingServicesWg.Add(1)
+
+		go func() {
+			defer supportingServicesWg.Done()
+
+			c := make(chan os.Signal, 1)
+			signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
+
+			select {
+			case <-ctx.Done():
+				f.logger.Debug("shutting down signal monitoring goroutine")
+				return
+			case s := <-c:
+				f.ErrorHandler.Fatal("user", fmt.Errorf("signal received: %v", s.String()))
```

Comment:
Hm where is this being handled? Where's the actual code that does the dumping to stdout?

Reply:
The ErrorHandler has always dumped the state inside the Fatal method. This code change just dumps it when a signal is received.